yaptide/celery/utils/pymc.py

```diff
@@ -10,118 +12,193 @@
     estimators_dict = {}
+
+    matching_files = list(dir_path.glob("*.bdo"))
+    if len(matching_files) == 0:
+        logging.error("No *.bdo files found in %s", dir_path)
+        return estimators_dict
+
+    logging.debug("Found %d *.bdo files in %s", len(matching_files), dir_path)
     files_pattern_pattern = str(dir_path / "*.bdo")
-    estimators_list = frompattern(files_pattern_pattern)
+    estimators_list = frompattern(pattern=files_pattern_pattern)
     for estimator in estimators_list:
         logging.debug("Appending estimator for %s", estimator.file_corename)
         estimators_dict[estimator.file_corename] = estimator
 
     return estimators_dict
 
 
-def run_fluka(dir_path: Path, task_id: str) -> dict:
-    """Function run in eventlet to run single fluka simulation"""
-    settings = SimulationSettings(input_path=dir_path,  # skipcq: PYL-W0612 # usefull
-                                  simulator_type=SimulatorType.fluka,
-                                  simulator_exec_path=None,  # useless
-                                  cmdline_opts="")  # useless
+def command_to_run_fluka(dir_path: Path, task_id: str) -> list[str]:
+    """Function to create command to run FLUKA."""
+    settings = SimulationSettings(
+        input_path=dir_path,  # skipcq: PYL-W0612 # usefull
+        simulator_type=SimulatorType.fluka,
+        simulator_exec_path=None,  # useless, we guess from PATH
+        cmdline_opts="")  # useless, not useful for fluka
+    update_rng_seed_in_fluka_file(dir_path, task_id)
+    command_as_list = str(settings).split()
+    command_as_list.append(str(dir_path))
+    return command_as_list
 
+
+def update_rng_seed_in_fluka_file(dir_path: Path, task_id: str) -> None:
+    """Function to update random seed in FLUKA input file."""
     input_file = next(dir_path.glob("*.inp"), None)
     if input_file is None:
         # if there is no input file, raise an error
         # this should never happen
         raise FileNotFoundError("Input file not found")
 
     class UpdateFlukaRandomSeed(Protocol):
-        """Protocol for updating random seed in fluka input file"""
+        """Definition of protocol for updating random seed in fluka input file.
+
+        Its purpose is to allow us to use private method of Runner class.
+        """
 
         def __call__(self, file_path: str, rng_seed: int) -> None:
             """Updates random seed in fluka input file"""
 
     random_seed = int(task_id.split("_")[-1])
-    # propably should be protected instead of private
     update_fluka_function: UpdateFlukaRandomSeed = Runner._Runner__update_fluka_input_file  # pylint: disable=W0212
     update_fluka_function(str(input_file.resolve()), random_seed)
 
-    command_as_list = str(settings).split()
-
-    logging.debug("Running fluka with setting %s", settings)
 
+def execute_fluka_process(dir_path: Path, command_as_list: list[str]) -> tuple[bool, str, str]:
```

Comment:
This looks pretty similar to execute_shieldhit_process, maybe these two can be merged ?

Reply:
Sure, I separated these functions to have simulator-specific logging but otherwise, they are pretty similar.